rules/no-useless-promise-resolve-reject.js

```diff
@@ -142,7 +142,27 @@ const create = context => {
 	return {
 		[selector](callExpression) {
 			const {functionNode, isInTryStatement} = getFunctionNode(callExpression);
-			if (!functionNode || !functionNode.async) {
+
+			if (!functionNode) {
+				return;
+			}
+
+			let isInPromiseThenOrCatchCallback = false;
+			if (
+				functionNode.parent
+				&& functionNode.parent.type === 'CallExpression'
+				&& functionNode.parent.callee.type === 'MemberExpression'
+				&& functionNode.parent.callee.property.type === 'Identifier'
+			) {
+				const {callee: {property}, arguments: arguments_} = functionNode.parent;
+				if (property.name === 'then') {
+					isInPromiseThenOrCatchCallback = arguments_[0] === functionNode || arguments_[1] === functionNode;
+				} else if (property.name === 'catch' || property.name === 'finally') {
+					isInPromiseThenOrCatchCallback = arguments_[0] === functionNode;
+				}
+			}
```

Comment:
Can you extract this to a function? maybe `isPromiseCallback(functionNode)`

Reply:
So we can `if (!functionNode || (!functionNode.async && !isPromiseCallback(functionNode)))`